db/hints/internal/hint_endpoint_manager.cc

```diff
@@ -53,46 +53,47 @@ bool hint_endpoint_manager::store_hint(schema_ptr s, lw_shared_ptr<const frozen_
             // We discard the future to this call to `with_gate()`, so wrapping this lambda
             // within `coroutine::lambda` will NOT work. That's why we need to copy these
             // variables by hand.
+            hint_endpoint_manager& self = *this;
             auto s = std::move(s_);
             auto fm = std::move(fm_);
             auto tr_state = std::move(tr_state_);
 
             ++_hints_in_progress;
             size_t mut_size = fm->representation().size();
-            shard_stats().size_of_hints_in_progress += mut_size;
+            self.shard_stats().size_of_hints_in_progress += mut_size;
 
             if (utils::get_local_injector().enter("slow_down_writing_hints")) {
                 co_await seastar::sleep(std::chrono::seconds(10));
             }
 
             try {
-                const auto shared_lock = co_await get_shared_lock(file_update_mutex());
+                const auto shared_lock = co_await get_shared_lock(self.file_update_mutex());
```

Comment:
@dawmd hi Dawid, i don't understand this change. what exactly the "captures" you want to keep alive? if it is `this`, then how can we extend the life cycle of `this` by holding a weak reference to it?

Reply:
`this` is a pointer, but it's still a variable and needs to be stored by the lambda. If the lambda object is freed, that memory will be freed as well and using `this`, implicitly or not, might lead to use-after-free. ASAN hasn't reported any issue with that so far, so maybe something more subtle is going on in this code, but that was my reasoning